Dynatrace/dynatrace-operator
# Avoid duplication of `--set-host-id-source` OneAgent argument

## Description

If the OneAgent argument `--set-host-id-source` is provided in the DynaKube, it causes a duplication for classic fullstack and host monitoring modes. It's then unclear which argument counts. 

This PR fixes the handling of this argument, so that it can be overridden in the DynaKube while still maintaining the applied default values otherwise.

## How can this be tested?

* deploy an ClassicFull stack Dynakube with and without `--set-host-id-source=fqdn` in it
* check the OneAgent pod specs, they should contain it only once (with value `fqdn` resp. `auto` as default)

## Checklist
